### PR TITLE
replace_latin boolean

### PR DESCRIPTION
**Pull request by eminbugrasaral**
Mentioned in my first merge request, smart_boolean enables to replace common latin letters with their ascii representations. 

Example:

    from slugify import slugify
    >>> slugify(u'Bakıcı geldi')
    u'bak\u0131c\u0131-geldi'
    >>> slugify(u'Bakıcı geldi', smart_replace=True)
    u'bakici-geldi'

So that people who does not have 'ı' letter in their keyboards will be able to match the string by using either 'ı' or 'i' from their keyboards. 

For current version, 'bakici' cannot match 'bakıcı'. But with smart_replace, 'bakıcı' and 'bakici' can both match 'bakıcı' because 'bakici' may represent 'bakıcı' within an English keyboard.

# List of common latin letters to replace

- ı, ì, í, î, ï -> i
-İ, Ì, Í, Î, Ï -> I
- ö, ó, ò, ô, õ, ø -> o
- Ö, Ò, Ó, Ô, Õ, Ø -> O
- ü, ù, ú, û -> u
- Ü, Ù, Ú, Û -> U
- à, á, â, ã, ä, å -> a
- À, Á, Â, Ã, Ä, Å -> A 
- æ -> ae
- Æ -> AE
- è, é, ê, ë -> e
- È, É, Ê, Ë -> E
- ñ -> n
- Ñ -> N
- ý, ÿ -> y
- Ý, Ÿ -> Y
- ş -> s
- Ş -> S
- ç -> c
- Ç -> C
- ğ -> g
- Ğ -> G

### CODE REVIEW

File: slugify/__init__.py [hunk 1]
```diff
@@ -26,7 +99,7 @@ def smart_text(s, encoding='utf-8', errors='strict'):
 SLUG_OK = '-_~'
 
 
-def slugify(s, ok=SLUG_OK, lower=True, spaces=False):
+def slugify(s, ok=SLUG_OK, lower=True, spaces=False, replace_latin=False):
```
Comment (davedash) [inline slugify/__init__.py]: I almost feel like replace_latin should be True by default.

If you wouldn't mind adding a docstring here that would be great too, at least explain what `replace_latin` does when enabled and when disabled.

File: slugify/tests.py [hunk 2]
```diff
@@ -33,11 +36,16 @@ def check(x, y):
          # I don't really care what slugify returns.  Just don't crash.
          (u'x𘍿', u'x'),
          (u'ϧ΃𘒬𘓣',  u'\u03e7'),
-         (u'¿x', u'x')]
+         (u'¿x', u'x'),
+         (u'Bakıcı geldi', u'bak\u0131c\u0131-geldi')]
+
+    replace_latin = [(u'Bakıcı geldi', u'bakici-geldi')]
```
Comment (davedash) [inline slugify/tests.py]: Maybe add a string with capital letters as well?

**Comment by davedash** [comment]
This looks good, if you can make those minor changes that'd be great, and I'll merge this.

**Comment by davedash** [comment]
Also... may want to consider this library too:

https://github.com/dimka665/awesome-slugify

It might do everything this one does and then some already.

**Comment by eminbugrasaral** [comment]
I checked that library but I liked this one since it's simple.

Made the changes.

**Comment by eminbugrasaral** [comment]
If it's possible to build a new version with that merge, I would like to

**Pull request closed without merging**